mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageResultIterator.java

```diff
@@ -209,93 +209,88 @@ public MailboxException getException() {
         private final MessageId messageId;
-
-        private long modSeq = -1;
-
         private final MailboxId mailboxId;
 
         public UnloadedMessageResult(MailboxMessage message, MailboxException exception) {
-            super();
-            internalDate = message.getInternalDate();
-            size = message.getFullContentOctets();
-            uid = message.getUid();
-            flags = message.createFlags();
-            modSeq = message.getModSeq();
+            messageMetaData = message.metaData();
             mailboxId = message.getMailboxId();
             messageId = message.getMessageId();
             this.exception = exception;
         }
 
         @Override
-        public MailboxId getMailboxId() {
-            return mailboxId;
+        public MessageMetaData messageMetaData() {
+            return messageMetaData;
         }
 
         @Override
-        public Flags getFlags() {
-            return flags;
+        public MessageUid getUid() {
+            return messageMetaData().getUid();
         }
 
         @Override
-        public Content getFullContent() throws MailboxException {
-            throw exception;
+        public MessageId getMessageId() {
+            return messageMetaData().getMessageId();
         }
 
         @Override
         public Date getInternalDate() {
-            return internalDate;
+            return messageMetaData().getInternalDate();
         }
 
         @Override
-        public Content getBody() throws MailboxException {
-            throw exception;
+        public Flags getFlags() {
+            return messageMetaData().getFlags();
+        }
+
+        @Override
+        public long getModSeq() {
+            return messageMetaData().getModSeq();
         }
 
         @Override
         public long getSize() {
-            return size;
+            return messageMetaData().getSize();
         }
 
         @Override
-        public MessageUid getUid() {
-            return uid;
+        public MailboxId getMailboxId() {
+            return mailboxId;
         }
 
         @Override
-        public MessageId getMessageId() {
-            return messageId;
+        public Content getFullContent() throws MailboxException {
+            throw exception;
+        }
+
+        @Override
+        public Content getBody() throws MailboxException {
+            throw exception;
         }
         
         @Override
         public int compareTo(MessageResult that) {
-            return uid.compareTo(that.getUid());
+            return getUid().compareTo(that.getUid());
         }
 
         @Override
-        public int hashCode() {
-            return Objects.hashCode(exception, internalDate, size, uid, flags, modSeq, messageId);
+        public final boolean equals(Object o) {
+            if (o instanceof UnloadedMessageResult) {
+                UnloadedMessageResult that = (UnloadedMessageResult) o;
+
+                return Objects.equals(this.exception, that.exception)

```

Comment:
you changed the actual behavior of the previous equals/hashcode. Now you compare exception, uid, messageId and mailboxId and previously it was size, uid, modseq, exception, internalDate and Flags.